app/models/guest.rb

```diff
@@ -0,0 +1,8 @@
+class Guest < ApplicationRecord
+  has_many :guest_rooms 
+  has_many :rooms, through: :guest_rooms
+
+  def self.unique_list_of_guests(id)
```

Comment:
This works.... but if you're passing in a Hotel ID, I would call your parameter name this as well, to distinguish it from a Guest ID (just a little confusing to read).